packages/flutter/test/widgets/system_context_menu_test.dart

```diff
@@ -895,4 +895,400 @@ void main() {
+      
+      state.hideToolbar();
+      await tester.pump();
+      expect(find.byType(SystemContextMenu), findsNothing);
+    },
+    skip: kIsWeb, // [intended]
+    variant: TargetPlatformVariant.only(TargetPlatform.iOS),
+  );
+
+  testWidgets(
+    'system context menu auto-closes after custom action',
+    (WidgetTester tester) async {
+      bool customActionCalled = false;
+      final TextEditingController controller = TextEditingController(text: 'test text');
+      addTearDown(controller.dispose);
+      
+      await tester.pumpWidget(
+        Builder(
+          builder: (BuildContext context) {
+            final MediaQueryData mediaQueryData = MediaQuery.of(context);
+            return MediaQuery(
+              data: mediaQueryData.copyWith(supportsShowingSystemContextMenu: true),
+              child: MaterialApp(
+                home: Scaffold(
+                  body: Center(
+                    child: TextField(
+                      controller: controller,
+                      contextMenuBuilder:
+                          (BuildContext context, EditableTextState editableTextState) {
+                            return SystemContextMenu.editableText(
+                              editableTextState: editableTextState,
+                              items: <IOSSystemContextMenuItem>[
+                                IOSSystemContextMenuItemCustom(
+                                  title: 'Test Action',
+                                  onPressed: () {
+                                    customActionCalled = true;
+                                  },
+                                ),
+                              ],
+                            );
+                          },
+                    ),
+                  ),
+                ),
+              ),
+            );
+          },
+        ),
+      );
+
+      await tester.tap(find.byType(TextField));
+      final EditableTextState state = tester.state<EditableTextState>(find.byType(EditableText));
+      expect(state.showToolbar(), true);
+      await tester.pump();
+      
+      expect(find.byType(SystemContextMenu), findsOneWidget);
+      
+      final SystemContextMenu menu = tester.widget<SystemContextMenu>(find.byType(SystemContextMenu));
+      final IOSSystemContextMenuItemCustom item = menu.items[0] as IOSSystemContextMenuItemCustom;
+      
+      final ByteData message = const StandardMethodCodec().encodeMethodCall(
+        MethodCall('SystemContextMenu.onPerformCustomAction', item.hashCode.toString()),
+      );
+      await TestDefaultBinaryMessengerBinding.instance.defaultBinaryMessenger.handlePlatformMessage(
+        'flutter/platform',
+        message,
+        (_) {},
+      );
+      
+      expect(customActionCalled, isTrue);
+      
+      // iOS system menus auto-close after custom actions on real devices,
+      // but in tests we need to manually call hideToolbar() to simulate this.
+      state.hideToolbar();
```

Comment:
Instead of calling hideToolbar would it be possible to send the `ContextMenu.onDismissSystemContextMenu` platform channel message in order to make this more realistic?